mobile/src/libs/components/input/input.tsx

```diff
@@ -27,24 +27,42 @@ const Input = <T extends FieldValues>({
   name,
   placeholder,
 }: Properties<T>): JSX.Element => {
+  const [isFocused, setIsFocused] = useState(false);
   const { field } = useFormController({ name, control });
 
   const { value, onChange, onBlur } = field;
 
   const error = errors[name]?.message;
   const hasError = Boolean(error);
 
+  const isFieldFilled = Boolean(value);
+
+  const handleFocus = (): void => {
+    setIsFocused(true);
+  };
+
+  const handleBlur = (): void => {
+    setIsFocused(false);
+    onBlur();
+  };
+
   return (
     <View>
-      <Text>{label}</Text>
+      <Text style={styles.label}>{label}</Text>
       <TextInput
         onChangeText={onChange}
         value={value}
-        onBlur={onBlur}
+        onFocus={handleFocus}
+        onBlur={handleBlur}
         placeholder={placeholder}
-        style={styles.input}
+        style={[
+          styles.input,
+          !hasError && (isFocused || isFieldFilled) && styles.filledInput,
+          hasError && styles.errorInput,
+        ]}
+        placeholderTextColor={styles.placeholder.color}
       />
-      <Text>{hasError && (error as string)}</Text>
+      <Text style={styles.errorText}>{hasError && (error as string)}</Text>
     </View>
   );
```

Comment:
for password fields we have to add secureTextEntry , so I recommend you to change the code for the input to this one: 
```
import React from 'react';
import {
  type Control,
  type FieldErrors,
  type FieldPath,
  type FieldValues,
} from 'react-hook-form';
import { TextInput } from 'react-native';

import { Text, View } from '#libs/components/components';
import { useFormController, useState } from '#libs/hooks/hooks';

import { styles } from './styles';

type Properties<T extends FieldValues> = {
  control: Control<T, null>;
  errors: FieldErrors<T>;
  label: string;
  name: FieldPath<T>;
  secure: boolean;
  placeholder: string;
};

const Input = <T extends FieldValues>({
  control,
  errors,
  label,
  name,
  secure,
  placeholder,
}: Properties<T>): JSX.Element => {
  const [isFocused, setIsFocused] = useState(false);
  const { field } = useFormController({ name, control });

  const { value, onChange, onBlur } = field;

  const error = errors[name]?.message;
  const hasError = Boolean(error);

  const isFieldFilled = Boolean(value);

  const handleFocus = (): void => {
    setIsFocused(true);
  };

  const handeBlur = (): void => {
    setIsFocused(false);
    onBlur();
  };

  return (
    <View>
      <Text style={styles.label}>{label}</Text>
      <TextInput
        onChangeText={onChange}
        value={value}
        onFocus={handleFocus}
        onBlur={handeBlur}
        secureTextEntry={secure}
        placeholder={placeholder}
        style={[
          styles.input,
          !hasError && (isFocused || isFieldFilled) && styles.filledInput,
          hasError && styles.errorInput,
        ]}
        placeholderTextColor={styles.placeholder.color}
      />
      <Text style={styles.errorText}>{hasError && (error as string)}</Text>
    </View>
  );
};

export { Input };
```